### PR TITLE
#167944683 User can vote and remove vote 

### PR DESCRIPTION
**Pull request by Dkabiswa**
#### What does this PR do?
Implements the vote product feature 
#### Description of Task to be completed?
- Add a vote on the product if it doesn't exist
- Removes vote of the product if it exists
#### How should this be manually tested?
- Clone this repo and checkout to this branch
- Run `npm run dev`
- Add vote via path in the documentation
- Remove vote using the same path 
- Remember `status code: 204` is a no content so remove vote returns no content
#### What are the relevant pivotal tracker stories?
[#167944683](https://www.pivotaltracker.com/story/show/167944683)
#### Screenshots (if appropriate)
Screen Shot 2019-08-22 at 18 33 17

#### Questions:
N/A

### CODE REVIEW

**Review by willywax — COMMENTED** [review]
Pulled the feature. It votes successfully. But when I hit the end point again to remove vote. It gives no response. 
Please check it out

**Comment by Dkabiswa** [comment]
@willywax, as I mentioned in the PR, 204 is a no content  status code whereby the server automatically does not reply with a body

File: backend/app.js [hunk 1]
```diff
@@ -16,6 +17,7 @@ app.get('/', (req, res) => {
 });
 
 app.use('/api/v1/auth', userRoutes);
+app.use('/api/v1/products', checkAuth, voteRoutes);
```
Comment (j0flintking02) [inline backend/app.js]: the feature looks good to me though was wondering if we could join these routes

Comment (Dkabiswa) [reply backend/app.js]: this means deleting the voteRoutes file and transferring the route to productsRoute File

**Review by willywax — APPROVED** [review]
Looks Good

File: backend/controllers/ProductController.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ class ProductController {
     const isOwner = await ProductService.checkOwner(req.params.id, id);
     if (!isOwner) {
       return res.status(401).json({
-        status: 401,
+        status: 402,
```
Comment (drayzii) [inline backend/controllers/ProductController.js]: Should be 401

**Review by drayzii — CHANGES_REQUESTED** [review]

**Review by drayzii — APPROVED** [review]

**Pull request merged**